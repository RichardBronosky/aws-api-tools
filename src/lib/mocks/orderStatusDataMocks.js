export const orderStatus = {
  AbleToReschedule: false,
  AdditionalDirections: '**** ALL DAY DELIVERY**** ****NO TIME FRAMES****',
  AffiliateData: '',
  Balance: '0.00',
  BillingAddress: {
    Address1: '1112 MARSH TRAIL CIR',
    Address2: '',
    City: 'ATLANTA',
    FirstName: 'SARAH',
    LastName: 'SMITH',
    State: 'GA',
    Zip: '30328',
  },
  Customer: {
    AllowText2AltPhoneNumber: false,
    AllowText2PhoneNumber: false,
    AltPhoneNumber: '',
    CMKey: '01*202/292-8491',
    CustomerID: 'SE006284958',
    Division: 'SE',
    FirstName: 'SARAH',
    LastName: 'SMITH',
    PhoneNumber: '202/292-8491',
  },
  DeliveryAddress: {
    Address1: '1112 MARSH TRAIL CIR',
    Address2: '',
    City: 'ATLANTA',
    Plus4: '',
    State: 'GA',
    Zip: '30328',
  },
  DeliveryCharge: '119.99',
  DeliveryDate: '2019-08-06T00:00:00.000Z',
  DeliveryType: 'E',
  Deposits: {
    Deposit: [
      {
        AuthorizationCode: '004547',
        AuthorizedAmount: '37.00',
        AuthorizedDate: '2019-08-04T00:00:00.000Z',
        CashCode: 'VA',
      },
    ],
  },
  Division: 'SE',
  EmailAddress: 'jamiamorrow@gmail.com',
  EstimatedDeliveryTimeFrame: '',
  FinDeposits: {
    Deposit: [
      {
        AuthorizationCode: '004547',
        AuthorizedAmount: '600.00',
        AuthorizedDate: '2019-08-04T00:00:00.000Z',
        CashCode: 'RGE',
      },
    ],
  },
  FinanceAccountNumber: '6019191263030424',
  FinancePlan: '90015-0121',
  FraudFlag: '',
  GCDeposits: {
    Deposit: [
      {
        AuthorizationCode: '004547',
        AuthorizedAmount: '1.95',
        AuthorizedDate: '2019-08-04T00:00:00.000Z',
        CashCode: 'GC',
      },
    ],
  },
  I95ind: false,
  LineItems: {
    LineItem: [
      {
        ChildItems: {
          LineItem: [
            {
              Quantity: 1,
              SelectedMonogramFee: 0,
              Sku: '36111639',
              UnitPrice: '193.24',
              UnitProtectionPrice: '0',
              UnitProtectionType: 'N',
            },
            {
              Quantity: 1,
              SelectedMonogramFee: 0,
              Sku: '36211631',
              UnitPrice: '250.76',
              UnitProtectionPrice: '0',
              UnitProtectionType: 'N',
            },
          ],
        },
        PackagePrice: '444',
        Quantity: 1,
        SelectedMonogramFee: 0,
        Sku: '3611163P',
        UnitPrice: '0',
        UnitProtectionPrice: '0',
        UnitProtectionType: 'N',
        DisplayName: 'Louie Gray Twin Loft Bed with Desk and Bookcase',
      },
      {
        ChildItems: null,
        Quantity: 1,
        SelectedMonogramFee: 0,
        Sku: '38096613',
        UnitPrice: '29',
        UnitProtectionPrice: '0',
        UnitProtectionType: 'N',
        DisplayName: 'Mattie Gray Chevron Bean Bag Chair',
      },
    ],
  },
  OrderDate: '2019-08-04T23:17:03.000Z',
  OrderMessage: 'OK',
  OrderNumber: '14000136I',
  OrderTotal: '638.95',
  PickUpData: {
    Phone: '',
    PickupAddress: { Address1: '', Address2: '', City: '', State: '', Zip: '' },
    PickupHours: { string: null },
    StoreName: '',
    StoreNumber: '980',
  },
  RemoteDeliveryDate: '0001-01-01T00:00:00.000Z',
  RemoteEstimatedDeliveryTimeFrame: '',
  ShipDate: '',
  Status: 'Open',
  StoreNumber: '980',
  TaxAmount: '45.96',
  TaxAreaId: '111211811',
  Active: true,
  DispatchTrack: 'https://fgo-rtg.dispatchtrack.com/track_order/f0d44a2/SE14000136I',
}

export const orderStatusList = [orderStatus]
