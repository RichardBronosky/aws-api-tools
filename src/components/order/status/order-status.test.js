import React from 'react'
import renderer from 'react-test-renderer'
import OrderStatus from './order-status'
import { orderStatus, orderStatusList } from '@mocks/orderStatusDataMocks'

describe('OrderStatus', () => {
  const testProps = {
    orderDetails: orderStatus,
    orderList: orderStatusList,
    setOrderStatusState: jest.fn(),
  }

  it('renders correctly with default props', () => {
    const tree = renderer.create(<OrderStatus { ...testProps } />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when orderDetails does not exist', () => {
    const tree = renderer.create(<OrderStatus { ...{ ...testProps, orderDetails: null } } />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when orderList does not exist', () => {
    const tree = renderer.create(<OrderStatus { ...{ ...testProps, orderList: null } } />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
