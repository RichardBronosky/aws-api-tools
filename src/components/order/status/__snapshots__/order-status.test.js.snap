// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderStatus renders correctly when orderDetails does not exist 1`] = `
<div
  className="order-status grid-container"
>
  <div
    className="cell small-12 card grid-x"
  >
    <div
      className="grid-x order-search"
    >
      <div
        className="cell small-12 medium-8 large-6 order-status-info"
      >
        <h1>
          View Order Status
        </h1>
        <p>
          Enter your order information to see your order details. You can pay an open balance, see your delivery date, and check your delivery timeframe. History is available for orders created within the last 180 days.
        </p>
        <form
          className="grid-x cell small-12"
        >
          <label
            className="cell small-12"
          >
            Order Number (ex: 12345678i)
          </label>
          <input
            className="cell small-12"
            id="order_status_number"
            maxLength={20}
            name="order_number"
            onChange={[Function]}
            type="text"
          />
          <label
            className="cell small-12"
          >
            Phone Number (ex: 8135551234)
          </label>
          <input
            className="cell small-12"
            name="order_phone_number"
            onChange={[Function]}
            type="tel"
          />
          <button
            className="primary"
            onClick={[Function]}
          >
            Find Order
          </button>
        </form>
      </div>
      <div
        className="cell small-12 medium-4 large-6 help-text"
      >
        <div
          className="background-box"
        >
          <h2>
            Need Help?
          </h2>
          <br />
          <p>
            Customer Care Center:
            <br />
            <a
              href="tel:(800) 766-6786"
            >
               (800) 766-6786
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="grid-x cell small-12 card active-orders-list"
    >
      <hr />
      <div
        className="order-list"
      >
        <div
          className="grid-x cell small-12"
        >
          <h2>
            Active Orders
          </h2>
          <div
            className="grid-x cell small-12 order-status-headings-top"
          >
            <div
              className="order-status-heading cell large-2"
            >
              Order #
            </div>
            <div
              className="order-status-heading cell large-2"
            >
              Order Placed
            </div>
            <div
              className="order-status-heading cell large-2"
            >
              Total
            </div>
            <div
              className="order-status-heading cell large-2"
            >
              Balance
            </div>
            <div
              className="order-status-heading cell large-2"
            >
              Status
            </div>
            <div
              className="order-status-heading cell large-2"
            />
          </div>
          <div
            className="grid-x cell small-12 order-status-orders"
          >
            <div
              className="grid-x cell small-12 order-status-order-group"
            >
              <div
                className="mobile left small-6"
              >
                 Order Number: 
              </div>
              <div
                className="order-status-order cell small-6 large-2"
              >
                 
                14000136I
              </div>
              <div
                className="mobile small-6 left"
              >
                 Order Placed: 
              </div>
              <div
                className="order-status-order cell small-6 large-2"
              >
                 
                8/4/2019
              </div>
              <div
                className="mobile small-6 left"
              >
                Total: 
              </div>
              <div
                className="order-status-order cell small-6 large-2"
              >
                 
                $638.95
              </div>
              <div
                className="mobile small-6 left"
              >
                Balance: 
              </div>
              <div
                className="order-status-order cell small-6 large-2"
              >
                 
                $0
              </div>
              <div
                className="mobile small-6 left"
              >
                Status: 
              </div>
              <div
                className="order-status-order cell small-6 large-2"
              >
                 
                Open
              </div>
              <div
                className="order-status-order cell small-12 large-2"
              >
                <button
                  className="primary"
                  onClick={[Function]}
                >
                  View Details
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`OrderStatus renders correctly when orderList does not exist 1`] = `
<div
  className="order-status grid-container"
>
  <div
    className="grid-x cell card small-12"
  >
    <hr />
    <div
      className="grid-x cell small-12 order-details"
    >
      <div
        className="grid-x cell small-12 order-status-detailed-header"
      >
        <div
          className="cell small-12 medium-6 large-6 order-number"
        >
          <h2>
            Order # 14000136I
          </h2>
          <p
            className="ordered-on"
          >
            Ordered on Sunday, August 04, 2019
          </p>
        </div>
        <div
          className="cell small-12 medium-6 large-6"
        >
          <button
            className="primary"
            onClick={[Function]}
          >
            &lt; Back to Orders
          </button>
        </div>
      </div>
      <div
        className="grid-x cell small-12 order-status-payment-history"
      >
        <div
          className="cell small-12"
        >
          <h3>
            Payment History
          </h3>
        </div>
        <div
          className="grid-x cell small-12 medium-6 large-6"
        >
          <div
            className="payment-section"
          >
            <div
              className="cell small-12"
            >
              <strong>
                Other Payments
              </strong>
            </div>
            <div
              className="cell small-12"
            >
              $38.95
               
              (
              <button
                aria-controls="other-payment-details-accordion"
                aria-expanded={false}
                className="other-payment-details-expander"
                onClick={[Function]}
                tabIndex="0"
              >
                see details
              </button>
              )
            </div>
            <div
              id="other-payment-details-accordion"
            />
          </div>
          <div
            className="payment-section"
          >
            <div
              className="cell small-12"
            >
              <strong>
                RTG Finance
              </strong>
            </div>
            <div
              className="cell small-12"
            >
              $600
               (
              <button
                aria-controls="finance-details-accordion"
                aria-expanded={false}
                className="finance-details-expander"
                onClick={[Function]}
                tabIndex="0"
              >
                see details
              </button>
              )
            </div>
            <div
              id="finance-details-accordion"
            />
          </div>
        </div>
        <div
          className="cell small-12 medium-6 large-6"
        >
          <span>
            <strong>
              Note: 
            </strong>
            Delivery dates on orders with an open balance are subject to change. Call Customer Care if you need help at 
            <a
              href="tel:(800)-766-6786"
            >
              (800)-766-6786
            </a>
            .
          </span>
        </div>
      </div>
      <div
        className="grid-x cell small-12 order-status-delivery-setup"
      >
        <div
          className="cell small-12"
        >
          <h3>
            Delivery & Setup
          </h3>
        </div>
        <div
          className="grid-x cell small-12 medium-6 large-6"
        >
          <div
            className="cell small-12 bold"
          >
            <strong>
              SHIPPED TO
            </strong>
          </div>
          <div
            className="cell small-12"
          >
            <p>
              SARAH
               
              SMITH
            </p>
            <div
              className="order-status-delivery-address"
            >
              <p>
                1112 MARSH TRAIL CIR
              </p>
              <p>
                
              </p>
              <p>
                ATLANTA
                , 
                GA
                , 
                30328
              </p>
            </div>
          </div>
        </div>
        <div
          className="cell small-12 medium-6"
        >
          <strong>
            Scheduled Delivery Date
          </strong>
          <p>
            Contact Customer Service to Schedule a Delivery Date - 
            <a
              href="tel:(800)-766-6786"
            >
              (800)-766-6786
            </a>
          </p>
        </div>
      </div>
      <div
        className="grid-x cell small-12 order-status-product-info"
      >
        <div
          className="grid-x cell small-12 order-product-info-list"
        >
          <div
            className="grid-x cell small-12 order-status-product-info-headings-main"
          >
            <div
              className="order-status-heading cell small-12 medium-4 large-7"
            >
              Product Information
            </div>
            <div
              className="order-status-heading cell small-12 medium-2 large-1"
            >
              Qty.
            </div>
            <div
              className="order-status-heading cell small-12 medium-3 large-2"
            >
              Price
            </div>
            <div
              className="order-status-heading cell small-12 medium-3 large-2"
            >
              Subtotal
            </div>
          </div>
          <div
            className="grid-x cell small-12 order-status-product-info-items"
          >
            <div
              className="grid-x cell small-12 order-product-info-list-item"
            >
              <div
                className="grid-x cell small-12 order-status-product-info-headings"
              >
                <div
                  className="order-status-heading cell small-12 medium-4 large-7 undefined"
                >
                  Louie Gray Twin Loft Bed with Desk and Bookcase
                  <p>
                    SKU: 3611163P
                  </p>
                </div>
                <div
                  className="order-status-heading cell small-12 medium-2 large-1"
                >
                  1
                </div>
                <div
                  className="order-status-heading cell small-12 medium-3 large-2"
                >
                  $444
                </div>
                <div
                  className="order-status-heading subtotal cell small-12 medium-3 large-2"
                >
                  $444
                </div>
              </div>
            </div>
            <div
              className="grid-x cell small-12 order-product-info-list-item"
            >
              <div
                className="grid-x cell small-12 order-status-product-info-headings"
              >
                <div
                  className="order-status-heading cell small-12 medium-4 large-7 undefined"
                >
                  Mattie Gray Chevron Bean Bag Chair
                  <p>
                    SKU: 38096613
                  </p>
                </div>
                <div
                  className="order-status-heading cell small-12 medium-2 large-1"
                >
                  1
                </div>
                <div
                  className="order-status-heading cell small-12 medium-3 large-2"
                >
                  $29
                </div>
                <div
                  className="order-status-heading subtotal cell small-12 medium-3 large-2"
                >
                  $29
                </div>
              </div>
            </div>
          </div>
          <div
            className="order-status-product-info-totals grid-x cell small-12"
          >
            <div
              className="grid-x cell small-12"
            >
              <div
                className="order-status-heading cell large-8"
              />
              <div
                className="order-status-heading cell small-9 large-2"
              >
                Sub Total
              </div>
              <div
                className="order-status-heading cell small-3 large-2"
              >
                $518.96
              </div>
            </div>
            <div
              className="grid-x cell small-12"
            >
              <div
                className="order-status-heading cell large-8"
              />
              <div
                className="order-status-heading cell small-9 large-2"
              >
                Delivery & Shipping
              </div>
              <div
                className="order-status-heading cell small-3 large-2"
              >
                $119.99
              </div>
            </div>
            <div
              className="grid-x cell small-12"
            >
              <div
                className="order-status-heading cell large-8"
              />
              <div
                className="order-status-heading cell small-9 large-2"
              >
                Tax
              </div>
              <div
                className="order-status-heading cell small-3 large-2"
              >
                $45.96
              </div>
            </div>
            <div
              className="grid-x cell small-12"
            >
              <div
                className="order-status-heading cell large-8"
              />
              <div
                className="order-status-heading cell small-9 large-2"
              >
                Total
              </div>
              <div
                className="order-status-heading cell small-3 large-2"
              >
                $638.95
              </div>
            </div>
            <div
              className="grid-x cell small-12 balance"
            >
              <div
                className="cell large-8"
              />
              <div
                className="order-status-heading cell small-9 large-2"
              >
                <strong>
                  Balance:
                </strong>
              </div>
              <div
                className="order-status-heading cell small-3 large-2"
              >
                <strong>
                  $0
                </strong>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div />
</div>
`;

exports[`OrderStatus renders correctly with default props 1`] = `
<div
  className="order-status grid-container"
>
  <div
    className="grid-x cell card small-12"
  >
    <hr />
    <div
      className="grid-x cell small-12 order-details"
    >
      <div
        className="grid-x cell small-12 order-status-detailed-header"
      >
        <div
          className="cell small-12 medium-6 large-6 order-number"
        >
          <h2>
            Order # 14000136I
          </h2>
          <p
            className="ordered-on"
          >
            Ordered on Sunday, August 04, 2019
          </p>
        </div>
        <div
          className="cell small-12 medium-6 large-6"
        >
          <button
            className="primary"
            onClick={[Function]}
          >
            &lt; Back to Orders
          </button>
        </div>
      </div>
      <div
        className="grid-x cell small-12 order-status-payment-history"
      >
        <div
          className="cell small-12"
        >
          <h3>
            Payment History
          </h3>
        </div>
        <div
          className="grid-x cell small-12 medium-6 large-6"
        >
          <div
            className="payment-section"
          >
            <div
              className="cell small-12"
            >
              <strong>
                Other Payments
              </strong>
            </div>
            <div
              className="cell small-12"
            >
              $38.95
               
              (
              <button
                aria-controls="other-payment-details-accordion"
                aria-expanded={false}
                className="other-payment-details-expander"
                onClick={[Function]}
                tabIndex="0"
              >
                see details
              </button>
              )
            </div>
            <div
              id="other-payment-details-accordion"
            />
          </div>
          <div
            className="payment-section"
          >
            <div
              className="cell small-12"
            >
              <strong>
                RTG Finance
              </strong>
            </div>
            <div
              className="cell small-12"
            >
              $600
               (
              <button
                aria-controls="finance-details-accordion"
                aria-expanded={false}
                className="finance-details-expander"
                onClick={[Function]}
                tabIndex="0"
              >
                see details
              </button>
              )
            </div>
            <div
              id="finance-details-accordion"
            />
          </div>
        </div>
        <div
          className="cell small-12 medium-6 large-6"
        >
          <span>
            <strong>
              Note: 
            </strong>
            Delivery dates on orders with an open balance are subject to change. Call Customer Care if you need help at 
            <a
              href="tel:(800)-766-6786"
            >
              (800)-766-6786
            </a>
            .
          </span>
        </div>
      </div>
      <div
        className="grid-x cell small-12 order-status-delivery-setup"
      >
        <div
          className="cell small-12"
        >
          <h3>
            Delivery & Setup
          </h3>
        </div>
        <div
          className="grid-x cell small-12 medium-6 large-6"
        >
          <div
            className="cell small-12 bold"
          >
            <strong>
              SHIPPED TO
            </strong>
          </div>
          <div
            className="cell small-12"
          >
            <p>
              SARAH
               
              SMITH
            </p>
            <div
              className="order-status-delivery-address"
            >
              <p>
                1112 MARSH TRAIL CIR
              </p>
              <p>
                
              </p>
              <p>
                ATLANTA
                , 
                GA
                , 
                30328
              </p>
            </div>
          </div>
        </div>
        <div
          className="cell small-12 medium-6"
        >
          <strong>
            Scheduled Delivery Date
          </strong>
          <p>
            Contact Customer Service to Schedule a Delivery Date - 
            <a
              href="tel:(800)-766-6786"
            >
              (800)-766-6786
            </a>
          </p>
        </div>
      </div>
      <div
        className="grid-x cell small-12 order-status-product-info"
      >
        <div
          className="grid-x cell small-12 order-product-info-list"
        >
          <div
            className="grid-x cell small-12 order-status-product-info-headings-main"
          >
            <div
              className="order-status-heading cell small-12 medium-4 large-7"
            >
              Product Information
            </div>
            <div
              className="order-status-heading cell small-12 medium-2 large-1"
            >
              Qty.
            </div>
            <div
              className="order-status-heading cell small-12 medium-3 large-2"
            >
              Price
            </div>
            <div
              className="order-status-heading cell small-12 medium-3 large-2"
            >
              Subtotal
            </div>
          </div>
          <div
            className="grid-x cell small-12 order-status-product-info-items"
          >
            <div
              className="grid-x cell small-12 order-product-info-list-item"
            >
              <div
                className="grid-x cell small-12 order-status-product-info-headings"
              >
                <div
                  className="order-status-heading cell small-12 medium-4 large-7 undefined"
                >
                  Louie Gray Twin Loft Bed with Desk and Bookcase
                  <p>
                    SKU: 3611163P
                  </p>
                </div>
                <div
                  className="order-status-heading cell small-12 medium-2 large-1"
                >
                  1
                </div>
                <div
                  className="order-status-heading cell small-12 medium-3 large-2"
                >
                  $444
                </div>
                <div
                  className="order-status-heading subtotal cell small-12 medium-3 large-2"
                >
                  $444
                </div>
              </div>
            </div>
            <div
              className="grid-x cell small-12 order-product-info-list-item"
            >
              <div
                className="grid-x cell small-12 order-status-product-info-headings"
              >
                <div
                  className="order-status-heading cell small-12 medium-4 large-7 undefined"
                >
                  Mattie Gray Chevron Bean Bag Chair
                  <p>
                    SKU: 38096613
                  </p>
                </div>
                <div
                  className="order-status-heading cell small-12 medium-2 large-1"
                >
                  1
                </div>
                <div
                  className="order-status-heading cell small-12 medium-3 large-2"
                >
                  $29
                </div>
                <div
                  className="order-status-heading subtotal cell small-12 medium-3 large-2"
                >
                  $29
                </div>
              </div>
            </div>
          </div>
          <div
            className="order-status-product-info-totals grid-x cell small-12"
          >
            <div
              className="grid-x cell small-12"
            >
              <div
                className="order-status-heading cell large-8"
              />
              <div
                className="order-status-heading cell small-9 large-2"
              >
                Sub Total
              </div>
              <div
                className="order-status-heading cell small-3 large-2"
              >
                $518.96
              </div>
            </div>
            <div
              className="grid-x cell small-12"
            >
              <div
                className="order-status-heading cell large-8"
              />
              <div
                className="order-status-heading cell small-9 large-2"
              >
                Delivery & Shipping
              </div>
              <div
                className="order-status-heading cell small-3 large-2"
              >
                $119.99
              </div>
            </div>
            <div
              className="grid-x cell small-12"
            >
              <div
                className="order-status-heading cell large-8"
              />
              <div
                className="order-status-heading cell small-9 large-2"
              >
                Tax
              </div>
              <div
                className="order-status-heading cell small-3 large-2"
              >
                $45.96
              </div>
            </div>
            <div
              className="grid-x cell small-12"
            >
              <div
                className="order-status-heading cell large-8"
              />
              <div
                className="order-status-heading cell small-9 large-2"
              >
                Total
              </div>
              <div
                className="order-status-heading cell small-3 large-2"
              >
                $638.95
              </div>
            </div>
            <div
              className="grid-x cell small-12 balance"
            >
              <div
                className="cell large-8"
              />
              <div
                className="order-status-heading cell small-9 large-2"
              >
                <strong>
                  Balance:
                </strong>
              </div>
              <div
                className="order-status-heading cell small-3 large-2"
              >
                <strong>
                  $0
                </strong>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div />
</div>
`;
